processor/src/coin/bitcoin.rs

```diff
@@ -0,0 +1,380 @@
+    let public_key = PublicKey::from_private_key(&secp, &private_key);
+    
+    let new_addr = Address::p2wpkh(&public_key, Network::Regtest).unwrap();
+    self.rpc.generate_to_address(1, new_addr.to_string().as_str()).await.unwrap();
+  }
+
+  #[cfg(test)]
+  async fn test_send(&self, address: Self::Address) {
+    use bitcoin::{Address, PrivateKey, PublicKey,
+        OutPoint, Sequence, Witness,util::sighash::SighashCache,
+        Script, PackedLockTime,EcdsaSighashType, Network, 
+        blockdata::transaction::{TxIn, TxOut, Transaction},
+        secp256k1::{rand, Secp256k1, Message, SecretKey}};
+
+    let secp = Secp256k1::new();
+    let secret_key = SecretKey::new(&mut rand::thread_rng());
+    let private_key = PrivateKey::new(secret_key, Network::Regtest);
+    let public_key = PublicKey::from_private_key(&secp, &private_key);
+    
+    let main_addr = Address::p2wpkh(&public_key, Network::Regtest).unwrap();
+
+    let mut vin_list = Vec::new();
+    let mut vout_list = Vec::new();
+
+    let new_block = self.get_latest_block_number().await.unwrap() + 1;
+    self.rpc.generate_to_address(1, main_addr.to_string().as_str()).await.unwrap();
+
+    for _ in 0..100 {
+      self.mine_block().await;
+    }
+
+    let block_number = new_block + 1;
+    let active_block = self.get_block(block_number).await.unwrap();
+    
+    let block_details = self.rpc.get_block(&active_block.1.block_hash()).await.unwrap();
+    let first_tx = &block_details.txdata[0];
+    let mut amount = 0;
+    for (index, output_tx) in first_tx.output.iter().enumerate() {
+      if output_tx.script_pubkey == main_addr.script_pubkey() {
+        vin_list.push(TxIn {
+          previous_output: 
+          OutPoint {
+              txid: first_tx.txid(),
+              vout: u32::try_from(index).unwrap(),
+          },
+          script_sig: Script::default(), 
+          sequence: Sequence(u32::MAX),
+          witness: Witness::default(), 
+        });
+        amount = output_tx.value;
+        vout_list.push(TxOut {
+          value: amount - 10000,
+          script_pubkey: address.script_pubkey(),
+        });
+      }
+    }
+
+    let mut new_transaction = Transaction {
+      version: 2,
+      lock_time: PackedLockTime(0),
+      input: vin_list,
+      output: vout_list,
+    };
+    
+    let mut sig_hasher = SighashCache::new(&new_transaction);
+    let script_code = Address::p2pkh(&public_key, Network::Testnet).script_pubkey();
+    let transactions_sighash = sig_hasher.segwit_signature_hash(0, &script_code, amount, EcdsaSighashType::All).unwrap();
+    let signed_der = secp.sign_ecdsa_low_r(&Message::from(transactions_sighash.as_hash()), &private_key.inner).serialize_der();
+    let mut signed_witness = Witness::new();
+    signed_witness.push_bitcoin_signature(&signed_der, EcdsaSighashType::All);
+    signed_witness.push(public_key.to_bytes());
+    new_transaction.input[0].witness = signed_witness;
```

Comment:
This *could* be simplified by using P2PKH. I don't care to ask you to move to an inferior format, from a working piece of code, in the name of simplicity, yet I did want to note the thought.